tap_hubspot/__init__.py

```diff
@@ -1086,6 +1149,8 @@ def main_impl():
     if args.state:
         STATE.update(args.state)
 
+    discover_additional_streams()
```

Comment:
Maybe directly calling `discover_contact_list_streams` is cleaner

Reply:
You might be right about it. I just thought that there might be some additional streams to discover in the future. And also I didn't want to have any specifics regarding business entities in the main function. So I leave this up to you: for me there's no issue in calling it directly.

Reply:
I noticed that it runs for discover and sync. Is this necessary to run in sync mode as well?

Reply:
I think it makes sense: we apply user selection to the list of available streams, the code looks nicer. 
Also it seems cheap enough to have it in both places.

Reply:
is it possible to not discover contact list in sync mode?

Reply:
Yes, it's possible. But there's a downside: the code that decides what streams to sync will become uglier (please refer to `get_streams_to_sync` and `get_selected_streams`) and will contain code specific to handling `contact_list_contacts` entity. Now it's a clean code that relies on the list of streams (`STREAMS` variable) and does not know anything specific to a particular stream

Reply:
What if the `discover` step outputs necessary contact list details in properties file, and in sync mode it only reads details from properties file and do further steps required for sync? Is this flow possible in this tap?

Reply:
Sunil, thanks again for the detailed explanation. Addressed. Please refer to `populate_streams_from_catalog()` method that populates streams without fetching them from the HubSpot server